#!/usr/bin/env python
PACKAGE = "human_obstacle"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "Whether to apply plugin or not", True)
gen.add("pointMinDiff", double_t, 0, "Maximum distance between observation to consider as the same object", 0.25, 0.0, 5.0)
gen.add("pointLifetime", double_t, 0, "Time after which the observation should be removed from current observation's list (diff between current and last seen time)", 0.5, 0.0 , 10.0)
gen.add("innerRadius", double_t, 0, "Radius of inner human area", 0.5, 0.0, 10.0)
gen.add("distanceStep", double_t, 0, "Distance between points added to costmap (best lower then resolution of costmap)", 0.03, 0.0, 1.0)
gen.add("radiusInnerStep", double_t, 0, "Angular step during creation of inner human area (in rads)", 0.01, 0.0, 1.0)
gen.add("outerRadius", double_t, 0, "Maximum radius of outer human area", 1.65, 0.0, 10.0)
gen.add("outerRoundnessA", double_t, 0, "First parameter of outer area", 2.0, 0.0, 1000.0)
gen.add("outerRoundnessB", int_t, 0, "Second parameter of outer area (MUST BE ODD NUMBER!)", 1, 0, 100)
gen.add("outerMaxAngle", double_t, 0, "Max angle of outer area (in rads)", 1.57079, 0.0, 6.283185)
gen.add("radiusOuterStep", double_t, 0, "Angular step during creation of outer human area (in rads)", 0.01, 0.0, 1.0)
gen.add("costDeminishSpeed", double_t, 0, "Defines how fast costs deminishing (inflates)", 3.5, 0.0, 1000.0)
gen.add("enablePublishingPeoplePositions", bool_t, 0, "Enable publishing poeple positions", True);

exit(gen.generate("human_obstacle", "human_layer", "human_layer"))
